test/test-server/cloud_mds_handlers.go

```diff
@@ -13,33 +12,39 @@ import (
 
 func (c *CloudRouter) HandleAllRolesRoute(t *testing.T) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		namespace := r.URL.Query().Get("namespace")
-		namespaces := strings.Split(namespace, ",")
-
-		var allRoles []mdsv2alpha1.Role
-		for _, ns := range namespaces {
-			switch ns {
-			case "ksql":
-				allRoles = append(allRoles, rbacKsqlRoles()...)
-			case "datagovernance":
-				allRoles = append(allRoles, rbacSRRoles()...)
-			case "dataplane":
-				allRoles = append(allRoles, rbacDataPlaneRoles()...)
-			case "public":
-				allRoles = append(allRoles, rbacPublicRoles()...)
-			case "streamcatalog":
-				allRoles = append(allRoles, rbacStreamCatalogRoles()...)
-			default:
-				allRoles = append(allRoles, rbacPublicRoles()...)
-			}
-		}
-
+		allRoles := generateAllRolesList()
 		allRolesResponse, _ := json.Marshal(allRoles)
 		_, err := w.Write(allRolesResponse)
 		require.NoError(t, err)
 	}
 }
 
+func generateAllRolesList() []mdsv2alpha1.Role {
+	var allRoles []mdsv2alpha1.Role
+	/*
+	 * This is the specific order in which roles are added to the list
+	 * during our mocked tests. It includes the same namespace multiple
+	 * times due to some legacy behavior, which prints the same role more
+	 * than once.
```

Comment:
> It includes the same namespace multiple times due to some legacy behavior, which prints the same role more than once.

Even though it's legacy behavior, and it sounds like you're trying to avoid making a breaking change, this is still a bug. Can we clean this up now?

Reply:
> This duplication bug only happens in our mocked integration tests.

Ah, got it!